tendrl/commons/flows/create_cluster/utils.py

```diff
@@ -178,3 +180,83 @@ def gluster_create_ssh_setup_jobs(parameters, skip_current_node=False):
         )
     return ssh_job_ids
 
+
+def accuire_node_lock(parameters, flow_name):
+    Event(
+        Message(
+            job_id=parameters['job_id'],
+            flow_id=parameters['flow_id'],
+            priority="info",
+            publisher=NS.publisher_id,
+            payload={
+                "message": "Checking if nodes are not locked already"
+            }
+        )
+    )
+    fail = False
+    for node in parameters['Node[]']:
+        key = "/nodes/%s/locked_by" % node
+        try:
+            job_info = NS._int.client.read(key).value
+            fail = True
+            job_info = job_info.encode('utf8').replace("'", "\"")
+            job_info = json.loads(job_info)
+            Event(
+                Message(
+                    job_id=parameters['job_id'],
+                    flow_id=parameters['flow_id'],
+                    priority="error",
+                    publisher=NS.publisher_id,
+                    payload={
+                        "message": "Node %s is already locked by a job %s" % \
+                            (
+                                node,
+                                job_info['job_id']
+                            )
+                    }
+                )
+            )
+        except EtcdKeyNotFound:
+            lock_info = dict(
+                job_id = str(parameters["job_id"]),
+                flow = flow_name
+            )
+            NS._int.client.write(key, lock_info)
```

Comment:
no wait if it is already exist then it is consider as already locked by some job, so control never come to this line so it is not required i feel